src/NLog/Targets/Wrappers/AsyncTargetWrapper.cs

```diff
@@ -358,27 +358,37 @@ protected virtual void StartLazyWriterTimer()
         /// </summary>
         /// <returns>Returns true when scheduled a timer-worker-thread</returns>
         protected virtual bool StartInstantWriterTimer()
+        {
+            return StartTimerUnlessWriterActive(true);
+        }
+
+        private bool StartTimerUnlessWriterActive(bool instant)
         {
             bool lockTaken = false;
             try
             {
                 lockTaken = Monitor.TryEnter(_writeLockObject);
                 if (lockTaken)
                 {
-                    // Lock taken means no timer-worker-thread is active writing, schedule timer now
-                    lock (_timerLockObject)
+                    // Lock taken means no other timer-worker-thread is trying to write, schedule timer now
+                    if (instant)
                     {
-                        if (_lazyWriterTimer != null)
+                        lock (_timerLockObject)
                         {
-                            // Not optimal to schedule timer-worker-thread while holding lock,
-                            // as the newly scheduled timer-worker-thread will hammer into the writeLockObject
-                            _lazyWriterTimer.Change(0, Timeout.Infinite);
-                            return true;
+                            if (_lazyWriterTimer != null)
+                            {
+                                // Not optimal to schedule timer-worker-thread while holding lock,
+                                // as the newly scheduled timer-worker-thread will hammer into the writeLockObject
+                                _lazyWriterTimer.Change(0, Timeout.Infinite);
+                                return true;
+                            }
                         }
                     }
+                    else
+                    {
+                        StartLazyWriterTimer();
```

Comment:
shouldn't we return `true` here?

Reply:
The return value is whether it successfully scheduled an instant timer, but no harm in returning true here.